cpp/include/cudf/io/text/data_chunk_source.hpp

```diff
@@ -36,11 +36,14 @@ namespace text {
  */
 class device_data_chunk {
  public:
+  virtual ~device_data_chunk()                     = 0;
```

Comment:
Default impl is fine here
```suggestion
  virtual ~device_data_chunk()                     = default;
```